src/orion/executor/multiprocess_backend.py

```diff
@@ -167,13 +167,18 @@ def __init__(self, n_workers=-1, backend="multiprocess", **kwargs):
         if n_workers <= 0:
             n_workers = multiprocessing.cpu_count()
 
+        self.pool_config = {"n_workers": n_workers, "backend": backend}
         self.pool = PoolExecutor.BACKENDS.get(backend, ThreadPool)(n_workers)
 
     def __setstate__(self, state):
-        self.pool = state["pool"]
+        log.warning("Nesting multiprocess executor")
+        self.pool_config = state["pool_config"]
+        backend = self.pool_config.get("backend", ThreadPool)
```

Comment:
Should this be 'thread' here instead?

Reply:
actually we do not need the default the line after it will take care of the default, 
I updated the code